balancer/rls/balancer.go

```diff
@@ -329,6 +324,14 @@ func (b *rlsBalancer) UpdateClientConnState(ccs balancer.ClientConnState) error
 		b.dataCache.resize(newCfg.cacheSizeBytes)
 		b.cacheMu.Unlock()
 	}
+	b.stateMu.Lock()
+	// Enqueue an event which will notify us when the above update has been
+	// propagated to all child policies, and the child policies have all
+	// processed their updates, and we have sent a picker update.
+	done := make(chan struct{})
+	b.updateCh.Put(resumePickerUpdates{done: done})
+	b.stateMu.Unlock()
+	<-done
```

Comment:
On second thoughts, it seems like we don't even have to acquire `stateMu` here. We are not touching any state guarded by `stateMu` here. We are only pushing a message on the `updateCh`. What do you think?